bin/ghe-restore

```diff
@@ -285,6 +285,16 @@ fi
 # Get GHES release version in major.minor format
 RELEASE_VERSION=$(ghe-ssh "$GHE_HOSTNAME" -- 'ghe-config --get core.package-version' | cut -d '.' -f 1,2)
 
+
+# If the backup being restored is from an appliance with Actions disabled, restoring it onto an appliance with Actions enabled will cause
+# mismatches in the secrets needed for Actions which ultimately results in Actions not working properly.
+ACTIONS_ENABLED_IN_BACKUP=$(git config -f $GHE_RESTORE_SNAPSHOT_PATH/settings.json --list | tr . _ | grep "app_actions_enabled" | awk -F"=" '{ print $2 }')
```

Comment:
@boxofyellow @conorsloan The `settings.json` file which contains the config I need has a structure like this:

```
[app "actions"]
    enabled = false
```

Which doesn't seem like JSON to me, but more like TOML or ini, unless JSON can in fact be written like this. The `git config` command above parses and lists the settings in the file. The "hierarchy" is then removed and replaced by key/value pairs in which the key has a structure like `<section-name>_<key`> (example: `app_actions_enabled`).

I am not sure this is the best way to read the settings or the best way to parse the file in bash and use the value I need, therefore I kindly request your advice 😄

Reply:
That might work Artifact Cache dealt with this problem like this.  Maybe something like that work work here too?

https://github.com/github/backup-utils/blob/11143678e531ba0e88af0e59db4db3b2ab986bf6/bin/ghe-restore#L292

Reply:
The one thing I would recommend here is to use parameter expansion for this, since it splits up the long chain of command and avoids use of `tr` and `awk` in favour of bash features. 

e.g. 
``` bash 
# Explicitly grabbing config in a familiar format with grep (outputs "app.actions.enabled=false")
var=$(git config -f $GHE_RESTORE_SNAPSHOT_PATH/settings.json --list || grep "app.actions.enabled") 
# Replace the .'s with _'s using replace (outputs "app_actions_enabled=false")
var=${var//./_}
# Remove everything before the = (outputs "false")
var=${var#*=}
```

You can use more descriptive names here or a temporary variable. e.g.
* assign repeatedly to ACTIONS_ENABLED_IN_BACKUP
* create a function and use a local variable 
* use a tmp variable to build the string and unset it 

This is personal preference more than a hard rule, but it tends to be easier to grok and debug than long pipelines.

Reply:
If that's possible, this might be a better solution than 4 different unix commands piped together.  `ghe-mssql-console` however seems to only be available on the appliance, how is `ghe-restore`  going to be able to query the database? For example, if I run `ghe-restore` on my machine, I should be able to just restore the backup I have locally.

Also, this line is running a query on the appliance, but we need to know if the backup we are restoring has Actions enabled or not. The status of actions on the appliance is something we can already retrieve with `ghe-ssh "$GHE_HOSTNAME" -- 'ghe-config --true app.actions.enabled'`, used below.

Reply:
At the very least I would:
* Grep before tr, since that matches the pattern actually present in the file 
* Consider using the second parameter expansion `${var#*=}` to remove everything before = instead of `awk`

Reply:
>ghe-mssql-console however seems to only be available on the appliance, how is ghe-restore going to be able to query the database?

Correct. But in artifactCache [here](https://github.com/github/backup-utils/blob/master/bin/ghe-restore#L292). We first logging to ghe instance (see in last ``ghe-ssh "$GHE_HOSTNAME" /bin/bash``) and then running ``ghe-mssql-console``

Reply:
@tiwarishub thanks for your comment! This seems to be a valid approach when we want to check if the appliance which we are restoring the backup onto has ArtifactCache enabled. In my case, I want to check if the backup that is being restored has Actions enabled, the other way around therefore. I see that for ArtifactCache we are also checking some `.bak` files, that might work here.

Reply:
Update: I wasn't able to find any obvious way to use data/files from the backup's database snapshot to tell if actions is enabled or not. Using the value in `settings.json` seems like the safest option to me at the moment. I will close this conversation but feel free to reopen it as needed!